### ISSUE TITLE
consider-using-f-string is triggered when the formatting arguments are implicit

### ISSUE THREAD

**Issue by sam-s** [PyCQA/pylint#8296]
### Bug description

the following code should produce no diagnostics:
```python
"https://github.com/PyCQA/pylint/issues/8296"

date = (2020,1,2,3,4,5)
print("%4d-%02d-%02d %02d:%02d:%02d" % date)

def test_fn(pa_l):
    "my doc"
    return ["Alpha: {alpha:.0%}, Power: {power:.0%}, Nobs={nobs:.1f}".format(**pa)
            for pa in pa_l]
```

### Command used

```shell
pylint a.py
```

### Pylint output

```shell
************* Module a
a.py:4: [C0209(consider-using-f-string), ] Formatting a regular string which could be a f-string
a.py:8: [C0209(consider-using-f-string), test_fn] Formatting a regular string which could be a f-string
```

### Expected behavior

no diagnostics: it is impossible to replace these string formatting ops with f-strings.

### Pylint version

```shell
pylint 2.15.10
astroid 2.13.5
Python 3.10.6 (main, Nov 14 2022, 16:10:14) [GCC 11.3.0]
```

### OS / Environment

ubuntu

**Comment by Pierre-Sassoulas** [comment]
You could do:
```python
print(f"{date[0]:4d}-{date[1]:02d}-{date[2]:02d} {date[3]:02d}:{date[4]:02d}:{date[5]:02d}")

def test_fn(pa_l):
    """my doc"""
    return [
        f"Alpha: {pa['alpha']:.0%}, Power: {pa['power']:.0%}, Nobs={pa['nobs']:.1f}"
        for pa in pa_l
    ]
```
Or maybe:
```python
date = (2020,1,2,3,4,5)
print(f"{2020:4d}-{1:02d}-{2:02d} {3:02d}:{4:02d}:{5:02d}")

def test_fn(pa_l):
    """my doc"""
    return [
        f"Alpha: {alpha:.0%}, Power: {power:.0%}, Nobs={nobs:.1f}"
        for alpha, power, nobs in pa_l
    ]
```

**Comment by sam-s** [comment]
Your workarounds result in more complex and less compact code.
Essentially `pylint` here is telling me to make my code worse.
Maybe you could reconsider?

**Comment by Pierre-Sassoulas** [comment]
This is not a check about terseness but a check about performance see https://pylint.readthedocs.io/en/latest/user_guide/messages/convention/consider-using-f-string.html and in particular https://twitter.com/raymondh/status/1205969258800275456.

**Comment by nickdrozd** [comment]
At least for the tuple-date example, there is an easy fix for the false positive. Not sure about the comprehension case, but I'll take a look.

**Comment by nickdrozd** [comment]
The second example seems fine to me, but I opened a PR for the first one. The suggested replacement is much less concise than the original, and to me it feels like bending over backwards to appease the over-opinionated linter. Ultimately it's a matter of opinion about what warnings should be emitted.

**Comment by Pierre-Sassoulas** [comment]
Like in #8224  the performance are better with tuple in this case:
```
python -m timeit 'date = (2020,1,2,3,4,5); e = "%4d-%02d-%02d %02d:%02d:%02d" % date'
500000 loops, best of 5: 450 nsec per loop

python -m timeit 'date = (2020,1,2,3,4,5); e = f"{date[0]:4d}-{date[1]:02d}-{date[2]:02d} {date[3]:02d}:{date[4]:02d}:{date[5]:02d}"'                                            
500000 loops, best of 5: 845 nsec per loop
```

**Comment by mbyrnepr2** [comment]
Another example from https://github.com/pylint-dev/pylint/issues/8864: 
```python
print("%08x-%04x-%04x-%04x-%02x%02x%02x%02x%02x%02x" % (1, 2, 3, 4, 5, 6, 7, 8, 9, 10))
print(f"{1:08x}-{2:04x}-{3:04x}-{4:04x}-{5:02x}{6:02x}{7:02x}{8:02x}{9:02x}{10:02x}")
```

**Comment by mbyrnepr2** [comment]
I'm inclined to think this should be a "won't fix".
`consider-using-f-string` is a convention message, its not hard and fast rule which one should always accept - its a recommendation and it can be disabled per instance or globally per project. 
Emitting the message such as in the above examples isn't wrong per-se although sometimes, depending on various factors, like the length of the overall string, for example, some may disagree with it. "Pylint isn't smarter than you".

**Comment by Pierre-Sassoulas** [comment]
I think when the perf is better (tuple/functional context) we should stop raising by default because for me this is a performance check first before being a convention check. But it also makes sense that we'd have a convention check without anything to argue about (if you prefer ``"%08x" % (1)`` to ``{1:08x}`` then you're free to disable. Let's not add a dozen options with an implementation involving super convoluted conditionals to support them. If we want to do both we need to separate this in two checks (one convention, one warning ?), or if we want to keep only one check we need to take a decision about the goal of the check (convention or performance).

**Comment by dgutson** [comment]
This is one of those "choose your battles" situation :)

I will just say that many times the non-f string looks much more readable (a clean separation) than intermixing variables and format. In pur case, the format string is a GUID and by looking at the string, I can clearly see what the format is.
My last 0.2 cents.
